saleor/graphql/core/fields.py

```diff
@@ -71,11 +72,23 @@ def __init__(self, type_, *args, **kwargs):
         )
         kwargs.setdefault(
             "first",
-            graphene.Int(description="Return the first n elements from the list."),
+            graphene.Int(
+                description=(
+                    "Retrieve the first n elements from the list. "
+                    "Note that the system only allows fetching "
+                    f"a maximum of {settings.GRAPHQL_PAGINATION_LIMIT} objects."
```

Comment:
Maybe would be worth adding `a maximum of X objects in single query`. 
Or:
`Note that Saleor allows fetching a maximum of X objects in a single query.`
Adding `single query` would be more meaningful as someone could understand `maximum of {settings.GRAPHQL_PAGINATION_LIMIT} objects.` as a system limitation for the number of objects that you can have & fetch from the system.